R/module_teal_with_splash.R

```diff
@@ -1,23 +1,56 @@
 # This file adds a splash screen for delayed data loading on top of teal
 
-#' UI to show a splash screen in the beginning, then delegate to [srv_teal()]
+#' `teal` module integration with splash screen
 #'
 #' @description `r lifecycle::badge("stable")`
 #'
-#' The splash screen could be used to query for a password to fetch the data.
-#' [init()] is a very thin wrapper around this module useful for end-users which
+#' Module offers the dynamic display that switches between a custom splash screen
+#' for delayed data loading and a default splash screen.This helps ensure smooth
+#' user interaction, particularly during the initial heavy data processing or
+#' when querying for a password to fetch data.
+#'
+#' @details
+#' [`init()`] is a wrapper around this module useful for end-users which
 #' assumes that it is a top-level module and cannot be embedded.
 #' This function instead adheres to the `shiny` module conventions.
 #'
-#' If data is obtained through delayed loading, its splash screen is used. Otherwise,
-#' a default splash screen is shown.
-#'
-#' Please also refer to the doc of [init()].
+#' @seealso [init()]
 #'
 #' @param id (`character(1)`)
 #'   module id
 #' @inheritParams init
+#' @param modules (`teal_modules`) object containing the output modules which
+#'   will be displayed in the `teal` application. See [modules()] and [module()] for
+#'   more details.
+#' @inheritParams shiny::moduleServer
+#' @return
+#' - `ui_teal_with_splash` returns `HTML` for `shiny` UI module.
+#' - `srv_teal_with_splash` returns `reactive` containing `teal_data` object when data is loaded.
+#' If data is not loaded yet, `reactive` returns `NULL`.
```

Comment:
I think describing the return value of the UI function is redundant, especially after mentioning `shiny` module convention. It's enough to say that the module's server functions returns whatever it returns.